packages/maskbook/src/components/InjectedComponents/NFT/NFTBadge.tsx

```diff
@@ -11,53 +13,72 @@ const useStyles = makeStyles()({
         display: 'flex',
         justifyContent: 'center',
         alignItems: 'center',
-        position: 'absolute',
-        left: 0,
-        top: 46,
-        right: 0,
+        transform: 'scale(0.5)',
+        paddingTop: 8,
+        paddingBottom: 8,
     },
     wrapper: {
-        background:
-            'linear-gradient(106.15deg, #FF0000 5.97%, #FF8A00 21.54%, #FFC700 42.35%, #52FF00 56.58%, #00FFFF 73.01%, #0038FF 87.8%, #AD00FF 101.49%, #FF0000 110.25%)',
-        borderRadius: 3,
-        minWidth: 43,
-        width: 'auto',
         display: 'flex',
         justifyContent: 'center',
+        flexDirection: 'column',
+        alignItems: 'center',
     },
     icon: {
-        width: 28,
-        height: 11,
+        width: 12,
+        height: 12,
+        transform: 'translate(0px, 2px)',
     },
     text: {
         fontSize: 10,
-        transform: 'scale(0.8)',
+        whiteSpace: 'nowrap',
         margin: 0,
         color: 'white',
-        whiteSpace: 'nowrap',
         textShadow:
             '1px 1px black, 1px 0px black, 0px 1px black, -1px 0px black, 0px -1px black, -1px -1px black, 1px -1px black, -1px 1px black',
-        lineHeight: 1,
-        textDecoration: 'none',
     },
     link: {
         display: 'flex',
         flexDirection: 'column',
         alignItems: 'center',
         AlignJustify: 'center',
     },
+    box: {
+        background:
+            'linear-gradient(106.15deg, #FF0000 5.97%, #FF8A00 21.54%, #FFC700 42.35%, #52FF00 56.58%, #00FFFF 73.01%, #0038FF 87.8%, #AD00FF 101.49%, #FF0000 110.25%)',
+        borderRadius: 4,
+    },
+    loading: {
+        width: 64,
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        height: '100%',
+    },
 })
 
 interface NFTBadgeProps extends withClasses<'root' | 'text' | 'icon'> {
     avatar: AvatarMetaDB
     size?: number
 }
+
+function formatPrice(amount: string) {
+    const _amount = new BigNumber(amount ?? '0')
+    if (_amount.isZero()) return '0'
+    if (_amount.isLessThan(1)) return _amount.toFixed(1, 2)
+    if (_amount.isGreaterThan(1e6)) return `${_amount.div(1e6).toFixed(1)}M`
+    if (_amount.isGreaterThan(1e3)) return `${_amount.div(1e3).toFixed(1)}K`
+    return _amount.toFixed(2)
```

Comment:
```suggestion
    if (_amount.isLessThan(1)) return _amount.toFixed(1, 2)
    if (_amount.isLessThan(1e3)) return _amount.toFixed(2)
    if (_amount.isLessThan(1e6)) return `${_amount.div(1e3).toFixed(1)}K`
    return `${_amount.div(1e6).toFixed(1)}M`
```

makes code more readable